src/server/transaction.cc

```diff
@@ -976,7 +978,7 @@ string Transaction::DEBUG_PrintFailState(ShardId sid) const {
 void Transaction::EnableShard(ShardId sid) {
   unique_shard_cnt_ = 1;
   unique_shard_id_ = sid;
-  shard_data_.resize(1);
+  shard_data_.resize(IsActiveMulti() ? shard_set->size() : 1);
```

Comment:
As today we have in UnlockMulti a check for 
DCHECK_EQ(shard_data_.size(), shard_set->size());
I understand that we excpect that this array size will be shard_set->size() when running multi command, so that we do not shrink it when running some command in multi. Therefore the fix now is to resize it to shard_set->size() if IsActiveMulti()

Reply:
@dranikpg please comment / approve

Reply:
It turns out I fixed the issue I mentioned already 🤦🏻🤦🏻🤦🏻🤦🏻 The resize is the only fix needed 👍🏻 
https://github.com/dragonflydb/dragonfly/pull/2279

Reply:
@adiholden  when IsActiveMulti() is true, is it true that shard_data is already non-empty before the resize?
I am trying to understand preconditions

Reply:
Yes, it's non-empty, as it was used for scheduling

Reply:
nit: 
I prefer `if (shard_data_.empty()) shard_data_.resize(1); ` as being less intrusive.

Reply:
The convention in tx code is what Adi used